lib/layer.js

```diff
@@ -41,7 +40,7 @@ function Layer (path, options, fn) {
   this.name = fn.name || '<anonymous>'
   this.params = undefined
   this.path = undefined
-  this.regexp = pathRegexp((opts.strict ? path : loosen(path)), this.keys, opts)
+  this.regexp = pathRegexp.match((opts.strict ? path : loosen(path)), opts)
```

Comment:
May also want to pass through `decode: decodeParam` if you want the 400 error behavior, the default `decode = decodeURIComponent` so there's no wrapper.